hoodie-common/src/main/java/com/uber/hoodie/common/util/CompactionUtils.java

```diff
@@ -97,4 +102,53 @@ public static HoodieCompactionOperation buildHoodieCompactionOperation(Compactio
   public static CompactionOperation buildCompactionOperation(HoodieCompactionOperation hc) {
     return CompactionOperation.convertFromAvroRecordInstance(hc);
   }
+
+  /**
+   * Get all pending compaction workloads along with their instants
+   *
+   * @param metaClient Hoodie Meta Client
+   */
+  public static List<Pair<HoodieInstant, HoodieCompactionWorkload>> getAllPendingCompactionWorkloads(
+      HoodieTableMetaClient metaClient) {
+    List<HoodieInstant> pendingCompactionInstants =
+        metaClient.getActiveTimeline().filterPendingCompactionTimeline().getInstants().collect(Collectors.toList());
+    return pendingCompactionInstants.stream().map(instant -> {
+      try {
+        HoodieCompactionWorkload workload = AvroUtils.deserializeCompactionWorkload(
+            metaClient.getActiveTimeline().getInstantDetails(instant).get());
```

Comment:
this is an additional FS open + read.. We should seriously think about immediately priotizitng the filesytemview caching after this, to limit the namenode calls

Reply:
+1

Reply:
https://github.com/uber/hudi/issues/433